### PR TITLE
doc: cover more cache modes in rados/operations/cache-tiering.rst

### PR DESCRIPTION
**Pull request by smithfarm**
Fixes: http://tracker.ceph.com/issues/14153
Signed-off-by: Nathan Cutler <ncutler@suse.com>

### CODE REVIEW

**Comment by smithfarm** [comment]
Tried to make it consistent with http://docs.ceph.com/docs/master/dev/cache-pool/#cache-mode

File: doc/rados/operations/cache-tiering.rst [hunk 1]
```diff
@@ -62,6 +62,28 @@ configure how this migration takes place. There are two main scenarios:
...
+  operations are forwarded to the base tier. When objects are updated in the
+  base tier, Ceph makes no attempt to sync these updates to the corresponding
+  objects in the cache. This mode is intended for read-only workloads that do
+  not require consistency to be enforced by the storage system.
+
+- **Read-forward** does not promote objects from the base tier on read.
+  In other words, read requests that miss in the cache are forwarded to the
+  base tier. Write operations on the cache are accepted and flushed to the base
+  tier. This mode is useful when the base and cache tiers are comparably fast
+  for reads, but there is asymmetric performance or endurance for writes.
```
Comment (liewegas) [inline doc/rados/operations/cache-tiering.rst]: I think we should not mention the 'forward' modes, and (ideally) we should remove them entirely.  They do not work right.  We can make exsiting cache tiers using it switch to read-proxy on upgrade.

**Comment by liewegas** [comment]
IIRC most of these modes are marked as experimental and you ahve to pass --force to enable them?  Maybe they should be flagged that way here?

**Comment by tchaikov** [comment]
@smithfarm ping?

**Comment by smithfarm** [comment]
@tchaikov Removing the forward modes entirely, and making existing cache tiers switch to read-proxy on upgrade, is beyond the scope of this PR. Care to open a companion PR?

**Comment by tchaikov** [comment]
@smithfarm agreed. could you address sage's comment before we merge this PR?

> A more obvious/stronger warning on read-only is warranted, though.. my guess is most users would either not think about it or expect readonly to mean a write evicts the object.

we could highlight this using 
```rst
.. warning::

```

or just highlight the "no" in 

> Ceph makes no attempt to sync these updates to the corresponding objects in the cache.

**Comment by gregsfortytwo** [comment]
If these comments are no longer good, we probably want new ones. @smithfarm? :)

**Comment by liewegas** [comment]
unstale

**Comment by smithfarm** [comment]
Sigh. Sorry this has taken so long.

**Comment by smithfarm** [comment]
Changelog:

* added more obvious/stronger warning on read-only
* dropped read-forward and forward - i.e. the documentation will not mention these modes at all, even though they are present in the code

Still to be done:

* determine whether `--force` is required and mention that, if it is

**Comment by smithfarm** [comment]
```
target192168000083:/home/ubuntu # ceph osd tier add backing tier
pool 'tier' is now (or already was) a tier of 'backing'
target192168000083:/home/ubuntu # ceph osd tier cache-mode tier none
set cache-mode for pool 'tier' to none
target192168000083:/home/ubuntu # ceph osd tier cache-mode tier writeback
set cache-mode for pool 'tier' to writeback
target192168000083:/home/ubuntu # ceph osd tier cache-mode tier forward
Error EPERM: 'forward' is not a well-supported cache mode and may corrupt your data.  pass --yes-i-really-mean-it to force.
target192168000083:/home/ubuntu # ceph osd tier cache-mode tier readonly
Error EPERM: 'readonly' is not a well-supported cache mode and may corrupt your data.  pass --yes-i-really-mean-it to force.
target192168000083:/home/ubuntu # ceph osd tier cache-mode tier readforward
Error EPERM: 'readforward' is not a well-supported cache mode and may corrupt your data.  pass --yes-i-really-mean-it to force.
target192168000083:/home/ubuntu # ceph osd tier cache-mode tier readproxy
set cache-mode for pool 'tier' to readproxy
```

**Comment by smithfarm** [comment]
Changelog:

* mention that `readonly` cache mode is experimental

@liewegas @tchaikov Please re-review?

**Comment by liewegas** [comment]
see https://github.com/ceph/ceph/pull/28944

**Pull request merged**